src/ConnectWalletButton.tsx

```diff
@@ -0,0 +1,59 @@
+import { useState } from 'react';
+import styles from './ConnectWalletButton.module.css';
+import { formatWalletAddress } from './utils/wallet';
+
+interface ConnectWalletButtonProps {
+  walletAddress: string;
+  connectWallet: () => Promise<void>;
+  disconnectWallet: () => void;
+}
+
+const ConnectWalletButton = ({
+  walletAddress,
+  connectWallet,
+  disconnectWallet,
+}: ConnectWalletButtonProps) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const displayedButtonText = walletAddress
+    ? formatWalletAddress(walletAddress)
+    : 'Connect Wallet';
+
+  const copyAddressToClipboard = async () => {
+    await navigator.clipboard.writeText(walletAddress);
+    setIsDropdownOpen(false);
+  };
+
+  const onDisconnectClick = () => {
+    disconnectWallet();
+    setIsDropdownOpen(false);
+  };
+
+  return (
+    <div
+      className={styles.walletButtonContainer}
+      onMouseEnter={() => walletAddress && setIsDropdownOpen(true)}
```

Comment:
Only allow the dropdown to be shown if user is connected - otherwise there's nothing to disconnect or copy.